runtime/vm/resolvefield.cpp

```diff
@@ -608,14 +611,26 @@ fieldOffsetsStartDo(J9JavaVM *vm, J9ROMClass *romClass, J9Class *superClazz, J9R
 {
 
 	Trc_VM_romFieldOffsetsStartDo_Entry( NULL, romClass, superClazz, flags );
+#ifdef J9VM_OPT_VALHALLA_VALUE_TYPES
+	/* save the classLaoder */
+	J9ClassLoader *classLoader = state->classLoader;
```

Comment:
Very much yuck. Perhaps investigate passing the loader as a parameter - it's more explicit.

Reply:
Discussed with @DanHeidinga and we agree - let's have all the input data as parameters to the functions - feel free to store it in the state for later use.

Reply:
With VTs as a possibility, we will always need to pass a classLoader in every caller, but will not need to run java code (assuming we prepopulate the hash table).